### PR TITLE
docs: Standardize the README

### PR DESCRIPTION
**Pull request by gnumoksha**
What:
Standardizes the README file providing a common language and an implementation link.

Why:
There are differences between PSR's READMEs in regarding language and the lacking of implementations references. So I've updated all READMEs using the https://github.com/php-fig/http-factory as base.

### CODE REVIEW

**Review by Jean85 — APPROVED** [review]
LGTM :+1: Waiting for editor's review

File: README.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
-HTTP Server Handler
-===================
+HTTP Server Request Handlers
```
Comment (shadowhand) [inline README.md]:
```suggestion
HTTP Server Request Handlers for Middleware
```

File: README.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+==============
 
-Provides the `RequestHandlerInterface` of [PSR-15][psr-15].
+This repository holds the `RequestHandlerInterface` related to [PSR-15 (HTTP Server Request Handlers)][psr-url].
 
-[psr-15]: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-15-request-handlers.md
+Note that this is not a Server Request Handler implementation of its own. It is merely the interface that describe a Server Request Handler.
+
+You can find [implementations][implementation-url] and [installation instructions][package-url] for the specification on the packagist.
```
Comment (shadowhand) [inline README.md]: Same changes as https://github.com/php-fig/http-factory/pull/9 should be applied here.

**Review by shadowhand — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -1,6 +1,12 @@
-HTTP Server Handler
-===================
+HTTP Server Request Handlers for Middleware
+==============
```
Comment (shadowhand) [inline README.md]: The length of this should match the line above, or it should be switched to `#`

**Review by shadowhand — CHANGES_REQUESTED** [review]

**Review by shadowhand — APPROVED** [review]

**Comment by shadowhand** [comment]
Thanks!

**Pull request merged**